org.eclipse.ice.commands/src/main/java/org/eclipse/ice/commands/RemoteCopyFileCommand.java

```diff
@@ -23,55 +25,149 @@
-	 * descriptions.
+	 * Function which sets the two paths, source and destination, to those given by
+	 * the arguments. The ConnectionConfiguration also gives the remote connection
+	 * configuration for setting up the ssh and sftp channels.
 	 * 
-	 * @param src
-	 * @param dest
+	 * @param src  - source file to be moved
+	 * @param dest - destination for source file to be moved to
 	 */
-	public RemoteCopyFileCommand(String src, String dest) {
-		source = Paths.get(src);
-		destination = Paths.get(dest);
+	public void setConfiguration(String src, String dest) {
+		source = src;
+		destination = dest;
 	}
 
+	/**
+	 * See {@link org.eclipse.ice.commands.Command#execute()}
+	 */
 	@Override
 	public CommandStatus execute() {
-		// TODO Auto-generated method stub
-		return null;
+		status = run();
+		return status;
 	}
 
+	/**
+	 * See {@link org.eclipse.ice.commands.Command#run()}
+	 */
 	@Override
 	protected CommandStatus run() {
-		// TODO Auto-generated method stub
-		return null;
+		ChannelSftp channel = null;
+		try {
+			// Open the channel and connect it
+			channel = (ChannelSftp) getConnection().getSession().openChannel("sftp");
+			channel.connect();
+
+			// Determine how to proceed given what kind of copy it is
+			if (copyType == 1) { // If move type is local -> remote, use put
+				logger.info("Copying file " + source + " to " + destination);
+				channel.put(source, destination);
+			} else if (copyType == 2) { // if move type is remote -> local, use get
+				logger.info("Copying file " + source + " to " + destination);
+				channel.get(source, destination);
+			} else if (copyType == 3) { // if move type is remote -> remote, call function
+				logger.info("Executing cp " + source + " to " + destination);
+				copyRemoteToRemote();
+			} else {
+				logger.info("Unknown handle type...");
+				status = CommandStatus.FAILED;
+				return status;
+			}
+
+		} catch (JSchException | SftpException e) {
+			logger.error("Failed to connect to remote host. Exiting.");
+			e.printStackTrace();
+			status = CommandStatus.FAILED;
+			return status;
+		}
+
+		status = CommandStatus.SUCCESS;
+		return status;
 	}
 
-	@Override
-	public CommandStatus cancel() {
-		// TODO Auto-generated method stub
-		return null;
+	private void copyRemoteToRemote() throws JSchException {
+		getConnection().setChannel(getConnection().getSession().openChannel("exec"));
+		// TODO - test with windows, cp probably won't work
+		// Make a copy command to execute
+		String command = "cp " + source + " " + destination;
+		// Set the command for the JSch connection
+		((ChannelExec) getConnection().getChannel()).setCommand(command);
+		// If the channel isn't connected, connect and run the command
+		if (!getConnection().getChannel().isConnected()) {
+			getConnection().getChannel().connect();
+		} else {
+			logger.error("Channel isn't connected and can't copy remote to remote...");
+			throw new JSchException();
+		}
```

Comment:
This style is needlessly complicated. It's much clearer to do:

```
if (getConnection().getChannel().isConnected()) {
    logger.error("Channel isn't connected and can't copy remote to remote...");
    throw new JSchException();
}

getConnection().getChannel().connect();
```

Reply:
I altered these were relevant to be clearer.